tools/ocamllibdep.mll

```diff
@@ -99,13 +99,17 @@ type dir = string option
 
 let add_directory add_file phys_dir =
   let open Unix in
+  let files =
+    try Sys.readdir phys_dir
+    with Sys_error msg -> Format.eprintf "Error: @[%s@].@\n%!" msg; exit 1
```

Comment:
It's a bit weird to exit from a function that is supposed to be a library function...

Reply:
You mean raising and exiting here rather signalling the error to the upper level?

We would have a better overview of all different errors that can pop up, and more uniform treatment of them, by lifting them up to the main entry point.

This should actually apply to `coqdep` and `coq_makefile`. I can do that quickly while at it, if there is an agreement that this is a good way to proceed.

Reply:
Maybe throw a `DirectoryNotFound` exception and catch at the toplevel?

Reply:
I pushed a version which unifies the treatment of errors. I'm not sure this is what you had in mind exactly. Please tell.